WordPress/Jetpack/Classes/System/JetpackWindowManager.swift

```diff
@@ -101,12 +101,17 @@ private extension JetpackWindowManager {
     func showLoadWordPressUI(schemeUrl: URL) {
         let actions = MigrationLoadWordPressViewModel.Actions()
         let loadWordPressViewModel = MigrationLoadWordPressViewModel(actions: actions)
-        let loadWordPressViewController = MigrationLoadWordPressViewController(viewModel: loadWordPressViewModel)
-        actions.primary = {
+        let loadWordPressViewController = MigrationLoadWordPressViewController(
+            viewModel: loadWordPressViewModel,
+            tracker: migrationTracker
+        )
+        actions.primary = { [weak self] in
+            self?.migrationTracker.track(.loadWordPressScreenOpenTapped)
             UIApplication.shared.open(schemeUrl)
         }
-        actions.secondary = { [weak self] in
-            loadWordPressViewController.dismiss(animated: true) {
+        actions.secondary = { [weak self, weak loadWordPressViewController] in
```

Comment:
Adding `[weak loadWordPressViewController]` to break a reference cycle between the secondary action handler and `loadWordPressViewController`.

Without the `weak` reference, the memory would look like this:

`loadWordPressViewController` -> `loadWordPressViewModel` -> `actions.secondary` -> `loadWordPressViewController`

_`a -> b` means `a` owns `b`_

Reply:
Thanks for spotting this! I'd had a hunch something in this area would cause a reference cycle but didn't go back to investigate.